devicetypes/smartthings/orvibo-motion-sensor.src/Orvibo-Motion-Sensor.groovy

```diff
@@ -0,0 +1,200 @@
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ *  Author : jinkang zhang / jk0218.zhang@samsung.com
+ *  Date : 2018-07-04
+ */
+import physicalgraph.zigbee.clusters.iaszone.ZoneStatus
+import physicalgraph.zigbee.zcl.DataType
+
+metadata {
+	definition(name: "Orvibo Motion Sensor", namespace: "smartthings", author: "SmartThings", runLocally: false, minHubCoreVersion: '000.017.0012', executeCommandsLocally: false, mnmn: "SmartThings", vid: "generic-motion") {
+		capability "Motion Sensor"
+		capability "Configuration"
+		capability "Battery"
+		capability "Refresh"
+		capability "Health Check"
+		capability "Sensor"
+
+		command "enrollResponse"
+
+		fingerprint inClusters: "0000,0001,0003,0500", outClusters: ""
+	}
+
+	simulator {
+		status "active": "zone status 0x0001 -- extended status 0x00"
+        for (int i = 0; i <= 100; i += 11) {
+			status "battery ${i}%": "read attr - raw: 2E6D01000108210020C8, dni: 2E6D, endpoint: 01, cluster: 0001, size: 08, attrId: 0021, encoding: 20, value: ${i}"
+		}
+	}
+
+	preferences {
+		section {
+			image(name: 'educationalcontent', multiple: true, images: [
+					"http://cdn.device-gse.smartthings.com/Motion/Motion1.jpg",
+					"http://cdn.device-gse.smartthings.com/Motion/Motion2.jpg",
+					"http://cdn.device-gse.smartthings.com/Motion/Motion3.jpg"
+			])
+		}
+		section {
+			input title: "Motion Timeout", description: "These devices don't report when motion stops, so it's necessary to have a timer to report that motion has stopped. You can adjust how long this is below.", displayDuringSetup: false, type: "paragraph", element: "paragraph"
+			input "motionStopTime", "number", title: "Seconds", range: "*..*", displayDuringSetup: false, defaultValue: 5
+		}
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name: "motion", type: "generic", width: 6, height: 4) {
+			tileAttribute("device.motion", key: "PRIMARY_CONTROL") {
+				attributeState "active", label: 'motion', icon: "st.motion.motion.active", backgroundColor: "#00A0DC"
+				attributeState "inactive", label: 'no motion', icon: "st.motion.motion.inactive", backgroundColor: "#cccccc"
+			}
+		}
+		valueTile("battery", "device.battery", decoration: "flat", inactiveLabel: false, width: 2, height: 2) {
+			state "battery", label: '${currentValue}% battery', unit: ""
+		}
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+
+		main(["motion"])
+		details(["motion","battery", "refresh"])
+	}
+}
+
+private List<Map> collectAttributes(Map descMap) {
+	List<Map> descMaps = new ArrayList<Map>()
+
+	descMaps.add(descMap)
+
+	if (descMap.additionalAttrs) {
+		descMaps.addAll(descMap.additionalAttrs)
+	}
+
+	return  descMaps
+}
+
```

Comment:
No initial state checking in installed() method. You need to do it to synchronize device state with platform